test/e2e/placement_apply_strategy_test.go

```diff
@@ -0,0 +1,294 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.goms.io/fleet/pkg/controllers/work"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+var _ = Describe("validating CRP when resources exists", Ordered, func() {
+	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+	annotationKey := "annotation-key"
+	annotationValue := "annotation-value"
+	workNamespaceName := fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess())
+
+	BeforeAll(func() {
+		By("creating work resources on hub cluster")
+		createWorkResources()
+	})
+
+	AfterAll(func() {
+		By("deleting created work resources on hub cluster")
+		cleanupWorkResources()
+
+		By("deleting created work resources on member cluster")
+		cleanWorkResourcesOnCluster(allMemberClusters[0])
+	})
+
+	Context("Test a CRP place objects successfully (client-side-apply)", Ordered, func() {
+		BeforeAll(func() {
+			ns := appNamespace()
+			ns.SetOwnerReferences([]metav1.OwnerReference{
+				{
+					APIVersion: "another-api-version",
+					Kind:       "another-kind",
+					Name:       "another-owner",
+					UID:        "another-uid",
+				},
+			})
+			ns.Annotations = map[string]string{
+				annotationKey: annotationValue,
+			}
+			By(fmt.Sprintf("creating namespace %s on member cluster", ns.Name))
+			Expect(memberCluster1EastProdClient.Create(ctx, &ns)).Should(Succeed(), "Failed to create namespace %s", ns.Name)
+
+			// Create the CRP.
+			strategy := &placementv1beta1.ApplyStrategy{AllowCoOwnership: true}
+			createCRP(crpName, strategy)
+		})
+
+		AfterAll(func() {
+			By(fmt.Sprintf("deleting placement %s", crpName))
+			cleanupCRP(crpName)
+		})
+
+		It("should update CRP status as expected", func() {
+			crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), allMemberClusterNames, nil, "0", false)
+			Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP %s status as expected", crpName)
+		})
+
+		// This check will ignore the annotation of resources.
+		It("should place the selected resources on member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+		It("should have annotations on the namespace", func() {
+			want := map[string]string{annotationKey: annotationValue}
+			Expect(validateAnnotationOfWorkNamespaceOnCluster(memberCluster1EastProd, want)).Should(Succeed(), "Failed to override the annotation of work namespace on %s", memberCluster1EastProdName)
+		})
+
+		It("can delete the CRP", func() {
+			// Delete the CRP.
```

Comment:
we delete crp in the afterAll block

Reply:
each test will delete the crp in the afterAll block in line 65 to make sure we have a clean test env.